apis/nucleus/src/components/listbox/components/ListBoxHeader/icon-utils.js

```diff
@@ -17,16 +18,24 @@ const createDimensionIconData = (dimInfo, app) => {
       };
     case dimensionTypes.cyclic:
       return {
-        icon: CyclicIcon,
+        icon: app ? ReloadIcon : CyclicIcon,
         tooltip: 'Listbox.Cyclic',
-        onClick: () => {
-          app
-            .getDimension(dimInfo.qLibraryId)
-            .then((dimensionModel) => {
-              dimensionModel.stepCycle(1);
-            })
-            .catch(() => null);
-        },
+        onClick: app
+          ? () => {
+              app
+                .getDimension(dimInfo.qLibraryId)
+                .then((dimensionModel) => {
+                  if (!dimensionModel.stepCycle) {
+                    // eslint-disable-next-line no-console
+                    console.log(
+                      "engine version or engine api spec version doesn't have support for function stepCycle"
+                    );
+                  }
+                  dimensionModel.stepCycle(1);
```

Comment:
you shouldn't run the stepCycle function if it doesn't exist, just log the message.

Reply:
Yes you are right, I missed the return.